File: problem_287.rb
class Solution
    def solve(s0, s1)
        l0 = s0.size
        l1 = s1.size
        return false if l0 - l1 != 1

        for i in 0..l1
            return s0[i+1..-1] == s1[i..-1] if s0[i] != s1[i]
        end
        return true
    end
end

File: problem_287.cpp
#include <string>

bool solve(std::string s0, std::string s1) {

    size_t l1 = s0.size(), l2 = s1.size();

    if(l1 - l2 != 1)
        return false;

    if(l2 == 0)
        return true;

    size_t i=0, j=0;
    while(i<l1 && j< l2){
        if(s0[i] == s1[j])
            j++;
        i++;
        if(i-j>1)
            return false;
    }
    return (i-j == 1 && s0[i] == s1[j]) || i==j;
}
